src/coll/mod.rs

```diff
@@ -153,18 +160,23 @@ impl Collection {
                     filter: Option<bson::Document>,
                     options: Option<DistinctOptions>)
                     -> Result<Vec<Bson>> {
-
-        let opts = options.unwrap_or_else(DistinctOptions::new);
-
         let mut spec = bson::Document::new();
         spec.insert("distinct", Bson::String(self.name()));
         spec.insert("key", Bson::String(String::from(field_name)));
-        if filter.is_some() {
-            spec.insert("query", Bson::Document(filter.unwrap()));
+
+        if let Some(filter_doc) = filter {
```

Comment:
Although using `is_some()` and `unwrap()` is technically safe, it's more idiomatic to use `if let Some(...)`, so I changed to this style here and a few other places.